homeassistant/components/yeelight/light.py

```diff
@@ -277,10 +277,10 @@ def effect_list(self):
     @property
     def color_temp(self) -> int:
         """Return the color temperature."""
-        temp = self._get_property('ct')
-        if temp:
-            self._color_temp = temp
-        return kelvin_to_mired(int(self._color_temp))
+        temp_in_k = self._get_property('ct')
+        if temp_in_k:
+            self._color_temp = kelvin_to_mired(int(temp_in_k))
```

Comment:
Why would we cache this? Why not just convert it each time ? Or just convert it inside the update method ?

Reply:
If it would be better, I can store original value and convert it each time. I've recreated what was before, without much of thinking about it ( https://github.com/home-assistant/home-assistant/commit/6d3c3ce449e3aa97604ffce6557dfb450161c418#diff-6b4ac627753ee21cde162f000848ba3bL380 ). It was originally stores converted. Main issue fixed here is skip conversion, and exception, when new color_temp is None.

Reply:
Guess it's fine.